hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/gas/CanonicalDispatchPrices.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.node.app.service.contract.impl.exec.gas;
+
+import static java.util.Objects.requireNonNull;
+import static java.util.stream.Collectors.toMap;
+
+import com.hedera.node.app.hapi.fees.pricing.AssetsLoader;
+import com.hedera.node.app.service.mono.pbj.PbjConverter;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.AbstractMap;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.EnumMap;
+import java.util.Map;
+import java.util.Objects;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+/**
+ * A component providing the canonical prices in tinycents for each type of transaction dispatch.
+ */
+@Singleton
+public class CanonicalDispatchPrices {
+    private final Map<DispatchType, Long> pricesMap = new EnumMap<>(DispatchType.class);
+
+    @Inject
+    public CanonicalDispatchPrices(@NonNull final AssetsLoader assetsLoader) {
```

Comment:
as noted in a previous comment there's no unit test for this constructor (which would be tested indirectly via `canonicalPriceInTinycents`)

Reply:
I believe it does exist now.